### PR TITLE
Events

### PR DESCRIPTION
**Pull request by alexander16108**
- ## Added events 
- ## Added logic to the calculator for calculation 
- ## implemented all required stacks

### CODE REVIEW

File: src/component/buttons.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+    super(props);
+    this.state = {};
+
+    this.click = this.click.bind(this);
+  }
+
+  click() {
+    const { text, onClick, Obj } = this.props;
+    onClick(Obj, text);
+  }
```
Comment (dasileker) [inline src/component/buttons.js]: - [ ] The event click is not responding to make a calcule operation.

oooo

File: src/component/Calculator.js [hunk 2]
```diff
@@ -1,43 +1,64 @@
...
+    const { calcObj } = this.state;
+
+    if (Object.keys(calcObj).length === 0 || (calcObj.total === null
+      && calcObj.next === null && calcObj.operation === null)) {
+      display = '0';
+    } else {
+      display = calcObj.next != null ? calcObj.next
+        : calcObj.next != null ? calcObj.total : calcObj.operation;
+    }
+
```
Comment (dasileker) [inline src/component/Calculator.js]: - [ ] like the other file the render and the click event are not working properly, please refactor the code.

pppp

**Review by dasileker — CHANGES_REQUESTED** [review]
### **Hi @alexander16108,** 

### To Highlight

- The coding is clean.
- linters passing.
- commit is easy to follow.

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by AshakaE — APPROVED** [review]
### Hi @alexander16108  🙋‍♀️, 

# STATUS APPROVED ✔
Your project is complete!  There is nothing else to say other than... it's time to merge it 🔗 🚢 
Congratulations! 🎉 

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers 🥂  and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**